src/Penrose/Substance.hs

```diff
@@ -716,14 +735,19 @@ subSeparate = foldr separate ([], [])
                              (LD x) -> (x : decls, constrs)
                              (LC x) -> (decls, x : constrs)
 
+-- | Convert a "declare and bind" to a "declare, then bind" as separate statements
+toSubCore :: [SubStmt'] -> [SubStmt]
+toSubCore stmts = concatMap toCoreStmt stmts
```

Comment:
This can be done without the extra `SubStmt'` type by pattern matching on `DeclBind` only.